cargo-dist/src/installer.ps1

```diff
@@ -16,53 +16,135 @@ param (
     [Parameter(HelpMessage = 'The version of the App')]
     [string]$app_version = '{{APP_VERSION}}',
     [Parameter(HelpMessage = 'The URL of the directory where artifacts can be fetched from')]
-    [string]$artifact_download_url = '{{ARTIFACT_DOWNLOAD_URL}}'
+    [string]$artifact_download_url = '{{ARTIFACT_DOWNLOAD_URL}}',
+    [Parameter(HelpMessage = 'The file path of a local dist-manifest.json to use.')]
+    [string]$local_manifest
 )
 
-function Install-Binary($install_args) {
+function Install-Binary() {
   $old_erroractionpreference = $ErrorActionPreference
   $ErrorActionPreference = 'stop'
 
   Initialize-Environment
 
-  # Platform info injected by cargo-dist
-  $platforms = {{PLATFORM_INFO}}
-
-  $fetched = Download "$artifact_download_url" $platforms
+  $fetched = Download "$artifact_download_url"
   # FIXME: add a flag that lets the user not do this step
-  Invoke-Installer $fetched "$install_args"
+  Invoke-Installer $fetched
 
   $ErrorActionPreference = $old_erroractionpreference
 }
 
-function Download($download_url, $platforms) {
-  # FIXME: make this something we lookup based on the current machine
-  $arch = "x86_64-pc-windows-msvc"
+function Get-TargetTriple() {
+  try {
+    # NOTE: this might return X64 on ARM64 Windows, which is OK since emulation is available.
+    # It works correctly starting in PowerShell Core 7.3 and Windows PowerShell in Win 11 22H2.
+    # Ideally this would just be
+    #   [System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture
+    # but that gets a type from the wrong assembly on Windows PowerShell (i.e. not Core)
```

Comment:
could you elaborate on this? (is this a Known Issue with an article?)

Reply:
I wrote down a little bit more about the problem I encountered here:

https://github.com/AustinWise/smeagol/blob/master/PowershellNotes.md#footnote-why-we-need-to-use-reflection-to-access-runtimeinformation

I'll do a little more research to see if there is a better solution. Reflection is a bit verbose.

Reply:
Oh wow, this is awesome, thanks! I'm willing to tolerate the weird reflection with this kind of documentation backing it.